src/sidebar/components/annotation-action-bar.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+
+const { withServices } = require('../util/service-context');
+const { isShareable, shareURI } = require('../util/annotation-sharing');
+
+const AnnotationActionButton = require('./annotation-action-button');
+const AnnotationShareControl = require('./annotation-share-control');
+
+/**
+ * A collection of `AnnotationActionButton`s in the footer area of an annotation.
+ */
+function AnnotationActionBar({
+  annotation,
+  isPrivate,
+  onDelete,
+  onEdit,
+  onFlag,
+  onReply,
+  groups,
+  permissions,
+  session,
+  settings,
+}) {
+  // Is the current user allowed to take the given `action` on this annotation?
+  const userIsAuthorizedTo = action => {
+    return permissions.permits(
+      annotation.permissions,
+      action,
+      session.state.userid
+    );
+  };
+
+  const showDeleteAction = userIsAuthorizedTo('delete');
+  const showEditAction = userIsAuthorizedTo('update');
+  // Anyone may flag an annotation except the annotation's author.
+  // This option is even presented to anonymous users
+  const showFlagAction = session.state.userid !== annotation.user;
+  const showShareAction = isShareable(annotation, settings);
+
+  const annotationGroup = groups.get(annotation.group);
+
+  return (
+    <div className="annotation-action-bar">
+      {showEditAction && (
+        <AnnotationActionButton icon="edit" label="Edit" onClick={onEdit} />
+      )}
+      {showDeleteAction && (
+        <AnnotationActionButton
+          icon="trash"
+          label="Delete"
+          onClick={onDelete}
+        />
+      )}
+      <AnnotationActionButton icon="reply" label="Reply" onClick={onReply} />
+      {showShareAction && (
+        <div>
```

Comment:
What is the wrapper `<div>` for here?

Reply:
Another good catch. At some point this may have been for styling—e.g. `AnnotationShareControl` may have been resetting `flex` in a way that caused a break, but this doesn't seem to be the case anymore. Removing.